### PR TITLE
[W5.2b][T08-1] Dang Huu

### PR DESCRIPTION
**Pull request by ndhuu**

### CODE REVIEW

**Review by quinnzzzzz — COMMENTED** [review]
good implementation of sort function, test case is valid. Many changes commited due to indentation might want to avoid? Overall good maintenance of OOP style.

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -156,6 +156,11 @@ Views all details of the 1st person in the results of the `find` command.
 Clears all entries from the address book. +
 Format: `clear`
 
+== Sorting addressbook by name : `sort`
+
```
Comment (RubaP) [inline docs/UserGuide.adoc:160]: Good job for updating the UserGuide

File: src/seedu/addressbook/commands/SortCommand.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import seedu.addressbook.common.Messages;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+
+/**
+ * Finds and lists all persons in address book whose name contains any of the argument keywords.
```
Comment (RubaP) [inline src/seedu/addressbook/commands/SortCommand.java:13]: Wrong comment!

File: test/java/seedu/addressbook/data/AddressBookTest.java [hunk 3]
```diff
@@ -128,6 +128,15 @@ public void clear() {
         assertTrue(isEmpty(defaultAddressBook.getAllPersons()));
     }
 
+    @Test
+    public void sort() throws Exception {
+        defaultAddressBook.sort();
```
Comment (RubaP) [inline test/java/seedu/addressbook/data/AddressBookTest.java:133]: Kudos for adding JUnit test case.

File: test/expected.txt [hunk 4]
```diff
@@ -290,6 +290,15 @@
 || 
 || 2 persons listed!
 || ===================================================
+|| Enter command: || [Command entered:  sort]
```
Comment (RubaP) [inline test/expected.txt:293]: Good that you have updated I/O test cases as well!

File: src/seedu/addressbook/parser/Parser.java [hunk 5]
```diff
@@ -73,33 +74,36 @@ public Command parseCommand(String userInput) {
 
         switch (commandWord) {
 
-        case AddCommand.COMMAND_WORD:
-            return prepareAdd(arguments);
+            case AddCommand.COMMAND_WORD:
```
Comment (RubaP) [inline src/seedu/addressbook/parser/Parser.java:77]: As a practice always make sure that your PR do not contain any uncessary changes.

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 6]
```diff
@@ -140,4 +148,19 @@ public boolean equals(Object other) {
...
+    public static Comparator<Person> PersonNameComparator = new Comparator<Person>() {
+
+        public int compare(Person s1, Person s2) {
+            String PersonName1 = s1.getName().toString().toUpperCase();
+            String PersonName2 = s2.toString().toUpperCase();
+
+            //ascending order
+            return PersonName1.compareTo(PersonName2);
+
+            //descending order
```
Comment (RubaP) [inline src/seedu/addressbook/data/person/UniquePersonList.java:163]: Avoid unnecessary changes & comments in your PR

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 7]
```diff
@@ -140,4 +148,19 @@ public boolean equals(Object other) {
                 || (other instanceof UniquePersonList // instanceof handles nulls
                         && this.internalList.equals(((UniquePersonList) other).internalList));
     }
+
+
+    /*Comparator for sorting the list by Student Name*/
+    public static Comparator<Person> PersonNameComparator = new Comparator<Person>() {
```
Comment (RubaP) [inline src/seedu/addressbook/data/person/UniquePersonList.java:154]: Good that you have added comment explaining the role of the comparator. However, I would suggest to additionally mention that it is case-insensitive since it is not trivial

**Review by RubaP — COMMENTED** [review]
Good job! Please review the comments and close the PR
